packages/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -273,6 +280,13 @@ export const GridRootStyles = styled('div', {
     [`& .${c.columnHeader}:focus, & .${c.cell}:focus`]: {
       outline: `solid ${t.palette.primary.main} 1px`,
     },
+    [`&.${c['root--noToolbar']} [aria-rowindex="1"] [aria-colindex="1"]`]: {
+      borderTopLeftRadius: 'var(--unstable_DataGrid-radius)',
+    },
+    [`&.${c['root--noToolbar']} [aria-rowindex="1"] [aria-colindex="${visibleColumns.length}"]`]: {
+      borderTopRightRadius:
+        'calc((1 - var(--DataGrid-hasScrollY)) * var(--unstable_DataGrid-radius))', // if (!hasScrollY) { style.borderTopRightRadius = radius }
```

Comment:
With the MacOS "When scrolling" scrollbar setting, the issue is only with the top-left demo:

https://github.com/user-attachments/assets/a025eb8e-f22b-48db-b48b-b32969ab32ec

I thought about utilizing the `--DataGrid-scrollbarSize` variable here, but couldn't find a way to do so.
But if we add a new 0/1 variable that would be set to 1 only if the scrollbar size is non-zero, that might work.
Thoughts?

Reply:
I handled it in javascript instead, could you test again and let me know if all looks good on macOS?